assets/js/components/Board/index.js

```diff
@@ -1,46 +1,109 @@
 import React, { Component } from 'react';
 import Square from '../Square'
+import Message from '../Message'
 import axios from 'axios'
 
 class Board extends Component {
     constructor(props) {
         super(props)
         this.state = {
-            moves: Array(9)
+            gameStatus: "",
+            moves: [],
+            currentPlayer: "",
+            message: "Player one,",
+            error: ""
         }
     }
 
-    selectSquare(position_number) {
+    componentDidMount() {
+        axios.get('/newgame')
+        .then((response) => {
+            let data = response.data;
+            let boardData = data.board;
+            let currentPlayer = data.current_player;
+            let gameStatus = data.game_status;
+            let reduceBoard = Object.values(boardData)
+            this.setState({
+                gameStatus: gameStatus,
+                moves: reduceBoard,
+                currentPlayer: currentPlayer,
+            })
+        }).catch(error => console.log(error))
+    }
+
+    getCurrentPlayer() {
+        if (this.state.currentPlayer == "X") {
+            this.setState({
+                currentPlayer: "O",
+                message: "Player two,"
+            })
+            return "X"
+        } 
+        else {
+            this.setState({
+                currentPlayer: "X",
+                message: "Player one,"
+            })
+            return "O"
+        }
+            
+    }
+
+    removeErrorMessage() {
+        this.setState({error: " "});
+    }
+
+    selectSquare(positionNumber) {
         const moves = this.state.moves.slice();
-        moves[position_number] = "X";
-        this.setState({
-            moves: moves
-        });
-        this.postNewMark()
+        console.log(moves)
+        if (moves[positionNumber] == null) {
```

Comment:
What happens if `moves[positionNumber]` returns undefined? Or empty string? Give this a read:

https://basarat.gitbooks.io/typescript/content/docs/javascript/null-undefined.html

Reply:
I just used a bang here if(!moves[positionNumber])...